apm-sniffer/bootstrap-plugins/jdk-threadpool-plugin/src/main/java/org/apache/skywalking/apm/plugin/define/HierarchyThreadPoolExecutorInstrumentation.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.define;
+
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.HierarchyMatch;
+
+public class HierarchyThreadPoolExecutorInstrumentation extends AbstractThreadPoolExecutorInstrumentation {
```

Comment:
Consider to use `LogicalOrMatch` to replace this and `ThreadPoolExecutorInstrumentation`?

Reply:
> Consider to use `LogicalOrMatch` to replace this and `ThreadPoolExecutorInstrumentation`?

I have tried to use `LogicalMatchOperation.or` before. but it accepts `IndirectMatch` class type, however, `NameMatch` is not subclass of `IndirectMatch`. I  replaced `NameMatch` with `MultiClassNameMatch` and verified it just now ,it works.

`LogicalMatchOperation.or(HierarchyMatch.byHierarchyMatch(ENHANCE_CLASS), MultiClassNameMatch.byMultiClassMatch(ENHANCE_CLASS))`

Reply:
Make sense.